strawberry/field.py

```diff
@@ -307,45 +310,97 @@ def type_params(self) -> List[TypeVar]:
             return self.type.type_params
         return []
 
+    def resolve_type(
+        self,
+        *,
+        type_definition: Optional[StrawberryObjectDefinition] = None,
+    ) -> Union[  # type: ignore [valid-type]
+        StrawberryType,
+        Type[WithStrawberryObjectDefinition],
+        Literal[UNRESOLVED],
+    ]:
+        # We return UNRESOLVED by default, which means this case will raise a
+        # MissingReturnAnnotationError exception in _check_field_annotations
+        resolved = UNRESOLVED
+        allow_generic = True
+
+        # We are catching NameError because dataclasses tries to fetch the type
+        # of the field from the class before the class is fully defined.
+        # This triggers a NameError error when using forward references because
+        # our `type` property tries to find the field type from the global namespace
+        # but it is not yet defined.
+        with contextlib.suppress(NameError):
+            # Prioritise the field type over the resolver return type
+            if self.type_annotation is not None:
+                resolved = self.type_annotation.resolve()
+            elif self.base_resolver is not None and self.base_resolver.type is not None:
+                # Handle unannotated functions (such as lambdas)
+                # Generics will raise MissingTypesForGenericError later
+                # on if we let it be returned. So use `type_annotation` instead
+                # which is the same behaviour as having no type information.
+                resolved = self.base_resolver.type
+                allow_generic = False
+
+        # If this is a generic field, try to resolve it using its origin's
+        # specialized type_var_map
+        if _is_generic(resolved):  # type: ignore
+            specialized_type_var_map = get_specialized_type_var_map(
+                cast(type, self.origin)
+            )
+            if specialized_type_var_map and isinstance(resolved, StrawberryType):
+                resolved = resolved.copy_with(specialized_type_var_map)
+            elif specialized_type_var_map and has_object_definition(resolved):
+                resolved = resolved.__strawberry_definition__.copy_with(
```

Comment:
can we add a test for this and the code below? 😊

Reply:
Oh, just noticed that this check is actually impossible to happen =P

The code below should actually be covered (except for the same check that it does here). Codecov says that no paths there are covered, but if I remove the code there a lot of tests break.

I'll try to refactor it to see if it gets fixed